### PR TITLE
Show sharing in menu in grid view

### PR DESCRIPTION
**Pull request by skjnldsv**
Capture d’écran_2019-03-14_17-57-11
Capture d’écran_2019-03-14_17-57-20
Capture d’écran_2019-03-14_17-57-32

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

### CODE REVIEW

File: apps/files/js/fileactionsmenu.js [hunk 1]
```diff
@@ -109,11 +112,12 @@
 				}
 				return orderA - orderB;
 			});
+			console.info(items);
```
Comment (juliushaertl) [inline apps/files/js/fileactionsmenu.js]: I'd say this should be removed.

File: apps/files/js/fileactionsmenu.js [hunk 2]
```diff
@@ -109,11 +112,12 @@
...
 			items = _.map(items, function(item) {
 				item.nameLowerCase = item.name.toLowerCase();
 				return item;
 			});
-
+console.info(items);
```
Comment (juliushaertl) [inline apps/files/js/fileactionsmenu.js]: Same here.

**Comment by jancborchardt** [comment]
Was just comparing with the right-click menu and also with our logical sorting – and I have to correct myself: Let’s put the "Share" action first in the list, _above_ "Favorites" rather than below. Since elsewhere it’s always the first action, even being outside of the menu. :)


Otherwise awesome! :tada:

**Review by jancborchardt — CHANGES_REQUESTED** [review]
As said let’s put Share as first entry in the menu (sorry for the confusion) then it’s :+1: :)

**Comment by skjnldsv** [comment]
> As said let’s put Share as first entry in the menu (sorry for the confusion) then it’s  :)

Above favorite?

**Comment by skjnldsv** [comment]
@jancborchardt done! :wink:

**Comment by MorrisJobke** [comment]
It now only shows "Shared" for me with the normal share icon instead of the little chain for public links or the avatar for user/group shares. 😢

**Comment by skjnldsv** [comment]
> It now only shows "Shared" for me with the normal share icon instead of the little chain for public links

I guess I missed some check :)

>  or the avatar for user/group shares.

But this is only on incoming shares, right?

**Comment by MorrisJobke** [comment]
> But this is only on incoming shares, right?

Ah - right.

**Review by jancborchardt — APPROVED** [review]
sure

**Comment by MorrisJobke** [comment]
### Status of [17072](https://drone.nextcloud.com/nextcloud/server/17072): failure

#### TESTS=jsunit
<details><summary>Show full log</summary>

```
PhantomJS 2.1.1 (Linux 0.0.0) LOG: 'JQMIGRATE: Migrate is installed, version 1.4.1'
PhantomJS 2.1.1 (Linux 0.0.0) LOG: 'Type files registered'
PhantomJS 2.1.1 (Linux 0.0.0) OCA.Files.FileActionsMenu tests rendering does not render inline actions FAILED
	Expected 1 to equal 0.
	apps/files/tests/js/fileactionsmenuSpec.js:117:69
PhantomJS 2.1.1 (Linux 0.0.0) OCA.Sharing.Util tests Share action icon shows owner name when owner is available FAILED
	Expected false to equal true.
	apps/files_sharing/tests/js/shareSpec.js:170:65
PhantomJS 2.1.1 (Linux 0.0.0) OCA.Sharing.Util tests Share action keep share text after updating reshare FAILED
	Expected false to equal true.
	apps/files_sharing/tests/js/shareSpec.js:392:65
PhantomJS 2.1.1 (Linux 0.0.0) OCA.Sharing.Util tests Share action keep share text after unsharing reshare FAILED
	Expected false to equal true.
	apps/files_sharing/tests/js/shareSpec.js:425:65
PhantomJS 2.1.1 (Linux 0.0.0): Executed 1072 of 1076 (4 FAILED) (skipped 4) (13.745 secs / 13.472 secs)

```
</details>

**Comment by MorrisJobke** [comment]
@skjnldsv Mind to fix the unit tests and conflicts?

**Comment by skjnldsv** [comment]
@skjnldsv in my todo for today :)

**Comment by skjnldsv** [comment]
Fixed

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

**Pull request merged**